rust/src/router/direct_transport.rs

```diff
@@ -533,25 +525,34 @@ impl DirectTransport {
         let subscription_handler = {
             let handlers = Arc::clone(&handlers);
 
-            channel.subscribe_to_notifications(id.into(), move |notification| {
-                match serde_json::from_slice::<Notification>(notification) {
-                    Ok(notification) => match notification {
-                        Notification::Trace(trace_event_data) => {
-                            handlers.trace.call_simple(&trace_event_data);
-                        } /*
-                           * TODO.
-                          Notification::Rtcp => {
-                              handlers.rtcp.call(|callback| {
-                                  callback(notification);
-                              });
-                          }
-                          */
-                    },
-                    Err(error) => {
-                        error!("Failed to parse notification: {}", error);
+            channel.subscribe_to_fbs_notifications(
+                id.into(),
+                move |notification| match notification.event().unwrap() {
+                    notification::Event::TransportTrace => {
+                        let Ok(Some(notification::BodyRef::TransportTraceNotification(body))) =
+                            notification.body()
+                        else {
+                            panic!("Wrong message from worker: {notification:?}");
+                        };
+
+                        let trace_event_data =
+                            transport::TraceNotification::try_from(body).unwrap();
+                        handlers
+                            .trace
+                            .call_simple(&TransportTraceEventData::from_fbs(trace_event_data));
                     }
-                }
-            })
+                    notification::Event::DirecttransportRtcp => {
+                        let Ok(Some(notification::BodyRef::RtcpNotification(_body))) =
+                            notification.body()
+                        else {
+                            panic!("Wrong message from worker: {notification:?}");
+                        };
+
+                        // TODO.
+                    }
+                    _ => unimplemented!(),
```

Comment:
Hm, what are the other variants?

Reply:
Notification event types live in the [FBS notification scope `Notification::Event`](https://github.com/versatica/mediasoup/blob/fbs_transport_events/worker/fbs/notification.fbs#L27). Since all notifications for `DirectTransport` are already handled here, receiving any other notification event for `DirectTransort` would be a bug. Would a `panic!()` fit better here?

Ideally notification/request/reponse types should live in their specific scope, but this is what the schema allows us to do. I would be happy to re-review that once after all Rust part has been ported to FBS.

Reply:
Hm, I don't like the design where notifications for all transports are located in the same enum, this is error-prone. Previously each transport had their own notifications. We can still share code with enum like this if contents is identical:
```rust
enum DirectTransportNotification {
    Rtcp(RtcpTransportNotification),
}
```

`RtcpTransportNotification` above can be shared between different transports for example.

Also I don't like `Directtransport` prefix, `DirectTransport` would have been nicer. Why did it change?

Reply:
> Also I don't like Directtransport prefix, DirectTransport would have been nicer. Why did it change?

This should be a typo in the PR and indeed must be fixed.

Reply:
> RtcpTransportNotification above can be shared between different transports for example.

I'll try the approach we had before.

>> Also I don't like Directtransport prefix, DirectTransport would have been nicer. Why did it change?

> This should be a typo in the PR and indeed must be fixed.

That's auto-generated code. Nothing we can freely choose.

Reply:
> That's auto-generated code. Nothing we can freely choose.

I don't believe this, `DirecttransportRtcp` has `Rtcp`, so it capitalizes some things, just not others.

Reply:
Auto-generated code from where? How does the source look like? AFAIK flatbuffers converts underscore syntax to camelCase.

Reply:
> I don't believe this, DirecttransportRtcp has Rtcp, so it capitalizes some things, just not others.

`DIRECTTRANSPORT_RTCP` is renamed to `DirecttransortRtcp`. If it the event in the schemas was `DIRECT_TRANSPORT_RTCP then we would have `DirectTransportRtcp`, but then in the worker we would be dealing with `DIRECT_TRANSPORT_RTCP`, and we consider DirectTransport a single word.

Reply:
IMHO we should adhere to this syntax:

- Use CamelCase when possible.
- When it's not possible (fbs requires lowcase and underscores) use xxx_xxx syntax.
- When forced to write in capital letters, use _ to separate those sub words that otherwise would be in capital letter in the cases above.

This is:

- DirectTransport, WebRtcTransport
- direct_transport, webrtc_transport (exception here I know)
- DIRECT_TRANSPORT, WEBRTC_TRANSPORT (exception here as well).

Reply:
But the problem is that we also want to have names like this:

DIRECTTRANSPORT_XXXX

which causes the above discussion...

Ok, my thoughts:

We should focus on what we can control, and forget about auto generated names.